### PR TITLE
openfortivpn 1.4.0 (new formula)

### PR DESCRIPTION
**Pull request by mnencia**
openfortivpn is a client for PPP+SSL VPN tunnel services. It spawns a pppd
process and operates the communication between the gateway and this process.

It is compatible with Fortinet VPNs.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/openfortivpn.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "openssl"
+
+  def install
+    system "aclocal"
+    system "autoconf"
+    system "automake", "--add-missing"
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl"].opt_lib}"
```
Comment (dunn) [inline Formula/openfortivpn.rb]: These paths can't be specified as arguments to `configure`?

File: Formula/openfortivpn.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  homepage "https://github.com/adrienverge/openfortivpn"
+  url "https://github.com/adrienverge/openfortivpn/archive/v1.3.1.tar.gz"
+  sha256 "cd499641e4c036cf0fc6891ce3676539c77de85a3eff9785c507ea2bd8ec3db0"
+
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
+  depends_on "openssl"
+
+  def install
+    system "aclocal"
```
Comment (dunn) [inline Formula/openfortivpn.rb]: Is there no `autogen.sh` script?

File: Formula/openfortivpn.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    system "automake", "--add-missing"
+    ENV.prepend "LDFLAGS", "-L#{Formula["openssl"].opt_lib}"
+    ENV.prepend "CPPFLAGS", "-I#{Formula["openssl"].opt_include}"
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+  test do
+    system bin/"openfortivpn", "--version"
```
Comment (dunn) [inline Formula/openfortivpn.rb]: A test that checks more functionality would be great!

File: Formula/openfortivpn.rb (hunk 1 above)
Comment (mnencia) [inline Formula/openfortivpn.rb]: I've checked, and unfortunately this is the only way with the current source.

File: Formula/openfortivpn.rb (hunk 3 above)
Comment (mnencia) [inline Formula/openfortivpn.rb]: This is a VPN software, I've no idea on how to check its functionality without connecting to a server. --help and --version are the only informative functions provided by openfortivpn

The only tests included in the source are about code formatting, so nothing useful to check if the software works.

File: Formula/openfortivpn.rb (hunk 2 above)
Comment (mnencia) [inline Formula/openfortivpn.rb]: No, there isn't

File: Formula/openfortivpn.rb (hunk 3 above)
Comment (dunn) [inline Formula/openfortivpn.rb]: Gotcha, thanks.

**Comment by dunn** [comment]
Would you mind asking the developers if they'd consider adding a `autogen.sh`-style script, and patching `configure` to allow specifying the OpenSSL path?

**Comment by mnencia** [comment]
I've made a pull request to openfortivpn to fix these issues. https://github.com/adrienverge/openfortivpn/issues/144

**Comment by JCount** [comment]
@mnencia If you would like, it may make the most sense to close this PR and open a new one for the new `1.4.0` release, but if you want to simply update this in place that is fine as well. Good work with the openfortivpn PR. 😄

**Comment by mnencia** [comment]
I've updated this pull request to 1.4.0 version

**Comment by dunn** [comment]
Thanks for contributing to Homebrew! Merged as 22d6d1d3090773f696f2b65871c5149de0531228.

**Comment by DimitriPapadopoulos** [comment]
@mnencia @dunn FYI the _autotools_ files in _openfortivpn_ have been changed again. They now make use of `pkg-config` which seems like the right thing to do on Linux and BSD systems. To pick an alternate OpenSSL library, you may have to define `PKG_CONFIG_PATH`. Does this work with Homebrew or do you need option `--with-openssl` back?

**Comment by mnencia** [comment]
It should work.
There are two other formula that are using that approach: `curl.rb` and `xmount.rb`
However I'd like to know what @dunn thinks from the style point of view.

**Comment by mrbaseman** [comment]
My colleague has reported that the new formula works like charm :+1: 
To my understanding it uses the 1.4.0 release whereas the master branch has switched to `pkg-config`. On the other hand I did not see the formula itself making use of either of these options. But probably the core brings in support for such configure options (maybe both variants?).

**Comment by mnencia** [comment]
@mrbaseman the change has not yet been released. What @DimitriPapadopoulos was asking is if this change is compatible with Homebrew or they need to modify it before the next release.

My answer is: The formula has to be updated, but the change is fine.

I don't know if @dunn wants to add anything.

**Comment by DimitriPapadopoulos** [comment]
@mnencia Have you had the chance to test _openfortivpn_ 1.5.0 yet?

Please note that _Autotools_ support for OpenSSL has been rewritten in _openfortivpn_ and hopefully standardized. The `--with-openssl` option is gone, you may use `PKG_CONFIG_PATH` instead.

**Comment by mnencia** [comment]
I've tested it briefly and everything looks ok.

**Comment by DimitriPapadopoulos** [comment]
Ooops... I just noticed recent commit bf53714da1 by **ilovezfs**.
Thank you for the feedback!

**Pull request closed without merging**